library/src/main/java/io/ona/kujaku/downloaders/MapBoxOfflineResourcesDownloader.java

```diff
@@ -213,6 +211,7 @@ public void onError(String error) {
 
             @Override
             public void onError(String error) {
+                Log.e(TAG, error);
                 if(onDownloadMapListener != null) {
                     onDownloadMapListener.onError(error, "");
```

Comment:
What's the difference between error reason and error message in the onDownloadMapListener. Should the error message in this onDownloadMapListener call be blank?

Reply:
In my opinion, 

Error reason is supposed to be like the code OR basically summarised Reason for the error. Error message is supposed to be like the whole explanation whether the stack trace is included and any other relevant detailed information.

It is used in two ways, the error_reason is programmatically useful, while the error_message is supposed to be human consumable. The interface exists with both because of listening to map download progress from MapBoxSDK which provides OfflineRegion.OfflineRegionObserver, which on error provides an OfflineRegionError basically containing a reason & message. I needed to have a callback that I could pass down to be called by the SDK callbacks without changing the information passed.

"Context passed is null" as a reason & message being blank should be handled by the third-party application to show a user friendly message -> Like a default ERROR message on their side instead of defining this. If we define this then the third-party app might not know which messages it can override. The ones provided by MapBox pertain to Server problems, connection problems or mapbox style not being found which might most likely be due to the user connecting an incorrect Mapbox account.

Any method call associated with downloading -> downloading or resuming a map download has both the error_reason & error_message because of the download updates error

Reply:
I will get back on this...

Reply:
Cool. Sounds OK. Could you please add the documentation in the interface. Also, if your intent with the "" is to show that nothing is being passed, isn't `null` then the right construct to use?